### PR TITLE
Change `runIndex` to `expInfoIndex` in MDEvent

### PR DESCRIPTION
**Pull request by stonecoldhughes**
**Description of work.**
Renamed all forms of "Run index" with "expInfoIndex" to indicate that it is the index of the associated experiment info object.

Linked Issue: https://code.ornl.gov/sns-hfir-scse/diffraction/single-crystal/single-crystal-diffraction/-/issues/210
PR ---> ornl-next: https://github.com/mantidproject/mantid/pull/31135

**To test:**
Successful compilation and passing of existing tests is sufficient to test this change

*There is no associated issue.*

*This does not require release notes* because **it is an internal refactor**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/DataObjects/inc/MantidDataObjects/MDEvent.h [hunk 1]
```diff
@@ -192,7 +192,7 @@ template <size_t nd> class DLLExport MDEvent : public MDLeanEvent<nd> {
   int32_t getDetectorID() const { return detectorId; }
 
   /** Sets the detectorId of this event
-   * @param id :: new runIndex value. */
+   * @param id :: new expInfoIndex value. */
```
Comment (jmborr) [inline Framework/DataObjects/inc/MantidDataObjects/MDEvent.h:195]: this should be "new detectorID value"

File: Framework/DataObjects/inc/MantidDataObjects/MDGridBox.tcc [hunk 2]
```diff
@@ -1568,18 +1568,18 @@ public:
  signal
  * @param Coord      :: vector of MD event coordinates, nd(number of dimensions)
  coordinates for each event
- * @param runIndex   :: vector of run  indexes for N events.
+ * @param expInfoIndex   :: vector of run  indexes for N events.
```
Comment (jmborr) [inline Framework/DataObjects/inc/MantidDataObjects/MDGridBox.tcc:1571]: vector of experiment-info indexes...

File: Framework/DataObjects/test/WorkspaceCreationTest.h [hunk 3]
```diff
@@ -92,7 +92,7 @@ void run_create_partitioned_with_instrument(const Parallel::Communicator &comm,
   TS_ASSERT_EQUALS(i.size(), expectedSize);
 }
 
-void run_indexInfo_legacy_compatibility_partitioned_workspace_failure(const Parallel::Communicator &comm) {
```
Comment (jmborr) [inline Framework/DataObjects/test/WorkspaceCreationTest.h:95]: changes in this file should be reverted. `run_indexInfo` is conceptually unrelated to `MDEvent.runIndex`

File: Framework/PythonInterface/plugins/algorithms/SNSPowderReduction.py [hunk 4]
```diff
@@ -485,7 +485,7 @@ def PyExec(self):  # noqa
             # END-FOR
         # ENDIF (Sum data or not)
 
-        for (samRunIndex, sam_ws_name) in enumerate(samwksplist):
+        for (samExpInfoIndex, sam_ws_name) in enumerate(samwksplist):
```
Comment (jmborr) [inline Framework/PythonInterface/plugins/algorithms/SNSPowderReduction.py:488]: All changes should be reverted in this file. `samRunIndex` is unrelated to `MDEvent.runIndex`

File: Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/PoldiDataAnalysis.py [hunk 5]
```diff
@@ -159,7 +159,7 @@ def runMainAnalysis(self, correlationSpectrum):
         outputWorkspaces.append(refinedPeaks)
         outputWorkspaces.append(fitPlots)
 
-        indexedPeaks, unindexedPeaks = self.runIndex(refinedPeaks)
+        indexedPeaks, unindexedPeaks = self.expInfoIndex(refinedPeaks)
```
Comment (jmborr) [inline Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/PoldiDataAnalysis.py:162]: All changes should be reverted in this file. `runIndex` in unrelated to `MDEvent.runIndex`

File: Framework/PythonInterface/test/python/plugins/algorithms/VesuvioTOFFitTest.py [hunk 6]
```diff
@@ -64,7 +64,7 @@ def test_single_run_produces_correct_output_workspace_index0_kfixed_no_backgroun
         self.assertTrue(self._equal_within_tolerance(expected_bin_index_spec2, bin_index_spec2))
 
 
-    def test_single_run_index0_kfixed_no_background_with_ties(self):
+    def test_single_expInfoIndex0_kfixed_no_background_with_ties(self):
```
Comment (jmborr) [inline Framework/PythonInterface/test/python/plugins/algorithms/VesuvioTOFFitTest.py:67]: This change should be reverted in this file. `run_index0` is unrelated to `MDEvent.runIndex`

**Review by jmborr — CHANGES_REQUESTED** [review]
Very very thorough changes ! :smiley:

File: Framework/Crystal/src/FindUBUsingIndexedPeaks.cpp [hunk 7]
```diff
@@ -119,22 +119,22 @@ void FindUBUsingIndexedPeaks::exec() {
         std::vector<V3D> run_fhkl_vectors;
         std::vector<V3D> run_hkl_vectors;
         std::vector<V3D> run_mnp_vectors;
-        size_t run_indexed = 0;
+        size_t expInfoIndexed = 0;
```
[the remark this answers is not in the document]
Comment (stonecoldhughes) [reply Framework/Crystal/src/FindUBUsingIndexedPeaks.cpp:122]: done!

File: Framework/DataObjects/inc/MantidDataObjects/MDEvent.h (hunk 1 above)
Comment (stonecoldhughes) [reply Framework/DataObjects/inc/MantidDataObjects/MDEvent.h:195]: done!

File: Framework/DataObjects/inc/MantidDataObjects/MDGridBox.tcc (hunk 2 above)
Comment (stonecoldhughes) [reply Framework/DataObjects/inc/MantidDataObjects/MDGridBox.tcc:1571]: done!

File: Framework/DataObjects/test/WorkspaceCreationTest.h (hunk 3 above)
Comment (stonecoldhughes) [reply Framework/DataObjects/test/WorkspaceCreationTest.h:95]: done!

File: Framework/PythonInterface/plugins/algorithms/SNSPowderReduction.py (hunk 4 above)
Comment (stonecoldhughes) [reply Framework/PythonInterface/plugins/algorithms/SNSPowderReduction.py:488]: done!

File: Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/PoldiDataAnalysis.py (hunk 5 above)
Comment (stonecoldhughes) [reply Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/PoldiDataAnalysis.py:162]: done!

File: Framework/PythonInterface/test/python/plugins/algorithms/VesuvioTOFFitTest.py (hunk 6 above)
Comment (stonecoldhughes) [reply Framework/PythonInterface/test/python/plugins/algorithms/VesuvioTOFFitTest.py:67]: done!

**Review by jmborr — APPROVED** [review]
Everything looks good

**Pull request merged**